File: typescript/src/model/types/wdl-map-type.ts
/** Map type nodes. */
import { WdlType, WdlTypeComponentType } from './wdl-type.js';

/** Models a map type such as `Map[String, Int]`. */
export class WdlMapType extends WdlType {
  /** Creates a map type from its key type, value type, and optional flag. */
  public constructor(
    private mapKeyType?: WdlType,
    private mapValueType?: WdlType,
    optional = false,
  ) {
    super(optional);
  }

  /** Returns the broad type family for this node. */
  public componentType(): WdlTypeComponentType {
    return WdlTypeComponentType.MAP;
  }

  /** Returns the map key type. */
  public keyType(): WdlType | undefined {
    return this.mapKeyType;
  }

  /** Sets the map key type. */
  public setKeyType(mapKeyType: WdlType | undefined): void {
    this.mapKeyType = mapKeyType;
  }

  /** Returns the map value type. */
  public valueType(): WdlType | undefined {
    return this.mapValueType;
  }

  /** Sets the map value type. */
  public setValueType(mapValueType: WdlType | undefined): void {
    this.mapValueType = mapValueType;
  }
}

File: java/src/main/java/com/myriad/wdl/model/types/WdlMapType.java
package com.myriad.wdl.model.types;

/**
 * Map type node.
 *
 * <p>This models WDL map types such as {@code Map[String, Int]}.
 */
public final class WdlMapType extends WdlType {
  private WdlType mapKeyType;
  private WdlType mapValueType;

  public WdlMapType() {
    super();
  }

  public WdlMapType(WdlType mapKeyType, WdlType mapValueType, boolean optional) {
    super(optional);
    this.mapKeyType = mapKeyType;
    this.mapValueType = mapValueType;
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.MAP;
  }

  /** Returns the map key type. */
  public WdlType keyType() {
    return mapKeyType;
  }

  /** Sets the map key type. */
  public void setKeyType(WdlType mapKeyType) {
    this.mapKeyType = mapKeyType;
  }

  /** Returns the map value type. */
  public WdlType valueType() {
    return mapValueType;
  }

  /** Sets the map value type. */
  public void setValueType(WdlType mapValueType) {
    this.mapValueType = mapValueType;
  }
}
